pkgs/tools/misc/ollama/default.nix

```diff
@@ -1,36 +1,55 @@
 { lib
+, stdenv
+, fetchpatch
 , buildGoModule
 , fetchFromGitHub
 , llama-cpp
+, darwin
 }:
 
-buildGoModule rec {
+let
+  version = "0.1.25";
+  llama_lib = (llama-cpp.override {
+      static = false;
+    }).overrideDerivation (oldAttrs: {
+      patches = [
+        (fetchpatch {
+          url = "https://github.com/ollama/ollama/raw/v${version}/llm/patches/01-cache.diff";
+          hash = "sha256-MgaSEMaXyRhmD5SaYABe8zthfgVeWK6IyeK4lHmj8yE=";
+        })
+        (fetchpatch {
+          url = "https://github.com/ollama/ollama/raw/v${version}/llm/patches/02-shutdown.diff";
+          hash = "sha256-5VX8PaN+Yh9erdRARkpUXauuiDMPaCqTBdKjDRpzMRc=";
+        })
+      ];
+    });
+in
+buildGoModule {
   pname = "ollama";
-  version = "0.1.17";
+  version = version;
 
   src = fetchFromGitHub {
     owner = "jmorganca";
     repo = "ollama";
     rev = "v${version}";
-    hash = "sha256-eXukNn9Lu1hF19GEi7S7a96qktsjnmXCUp38gw+3MzY=";
+    hash = "sha256-qrmG+wI3FgQegoW/vgUKOZIpcoVOvPB939+dtIfIIq0=";
   };
 
-  patches = [
-    # disable passing the deprecated gqa flag to llama-cpp-server
-    # see https://github.com/ggerganov/llama.cpp/issues/2975
-    ./disable-gqa.patch
-
-    # replace the call to the bundled llama-cpp-server with the one in the llama-cpp package
-    ./set-llamacpp-path.patch
-  ];
+  buildInputs = [ llama_lib ]
+    ++ llama_lib.buildInputs
```

Comment:
I'm not quite sure why this is necessary. That's... not what shared libraries normally need?